### PR TITLE
REVIEW

### PR DESCRIPTION
**Pull request by acocac**
Hi 👋

Thanks authors and reviewers to agree on collaboratively improving the submitted notebook to EDS book ✨

The review PR aims to facilitate a detailed discussion of how to improve the notebook via ReviewNB, a third-party plugin in GitHub for displaying and commenting Jupyter Notebooks. Find further details [here](https://edsbook.org/notebooks/about/notebooks-technologies.html#reviewnb). For general comments, please use the REVIEW issue containing the checklist.

Authors: please reply to reviewers' comments in a timely manner and push changes from your repository to this PR. You'll see ReviewNB easily renders Jupyter Notebook rich diffs than GitHub (see [here](https://blog.reviewnb.com/github-not-rendering-interactive-notebook-widgets/)).

### CODE REVIEW

File: notebook.ipynb [hunk 1]
```diff
@@ -2,7 +2,15 @@
  "cells": [
```
Comment (weiji14) [inline notebook.ipynb:2]: <ul><li>Acronym API needs to be spelled out on first mention.</li><li>No citation or link to U-Net model's paper. Also inconsistent hyphenation of U-Net vs UNet.</li></ul>
<br/>

 _Reply via <a href='https://app.reviewnb.com/eds-book-gallery/67a1e320-7c47-4ea9-8df8-e868326bc90b/pull/5/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='1'/>

Comment (weiji14) [inline notebook.ipynb:2]: <blockquote>where sea ice does not form, land regions, and the <a href="https://blogs.egu.eu/divisions/cr/2016/10/14/image-of-the-week-the-polar-hole/" target="_blank">polar hole</a>.</blockquote><p>Sea ice does form over the polar hole (in the Arctic), we just don't observe it.</p><p> </p><p> </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/eds-book-gallery/67a1e320-7c47-4ea9-8df8-e868326bc90b/pull/5/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='9'/>

Comment (weiji14) [inline notebook.ipynb:2]: <p>Need to spell out ERA5 and ORA5 acronyms on first mention.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/eds-book-gallery/67a1e320-7c47-4ea9-8df8-e868326bc90b/pull/5/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='12'/>

Comment (weiji14) [inline notebook.ipynb:2]: <p>Users could benefit from knowing what the var_names <code>tas</code>,<code>zg</code>,<code>uas</code>,<code>vas</code> means, and it would be nice to link to a data dictionary somewhere (so that others can find different variables). I usually recommend <a href="https://docs.google.com/spreadsheets/d/1UUtoz6Ofyjlpx5LdqhKcwHFz2SGoTQV2_yekHyMfL9Y/edit" target="_blank">https://docs.google.com/spreadsheets/d/1UUtoz6Ofyjlpx5LdqhKcwHFz2SGoTQV2_yekHyMfL9Y/edit</a> for CMIP6, but there might be a similar one for ERA5.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/eds-book-gallery/67a1e320-7c47-4ea9-8df8-e868326bc90b/pull/5/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='13'/>

Comment (weiji14) [inline notebook.ipynb:2]: <p>Need to spell out OSI SAF on first instance. See <a href="https://osi-saf.eumetsat.int/about-us" target="_blank">https://osi-saf.eumetsat.int/about-us</a>, should be Ocean Sea Ice Satellite Application Facilities?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/eds-book-gallery/67a1e320-7c47-4ea9-8df8-e868326bc90b/pull/5/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='16'/>

Comment (weiji14) [inline notebook.ipynb:2]: <ul><li>Need to spell out CLI acronym on first instance.</li><li>Maybe link to <a href="https://github.com/icenet-ai/icenet-pipeline?tab=readme-ov-file#configuration" target="_blank">https://github.com/icenet-ai/icenet-pipeline?tab=readme-ov-file#configuration</a> for the configuration details</li><li>Need a link to what Airflow is. Assuming that you mean Apache Airflow.</li></ul>
<br/>

 _Reply via <a href='https://app.reviewnb.com/eds-book-gallery/67a1e320-7c47-4ea9-8df8-e868326bc90b/pull/5/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='39'/>

Comment (weiji14) [inline notebook.ipynb:2]: <p>Again, check that UNet is consistently hyphenated as U-Net or UNet througout the notebook.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/eds-book-gallery/67a1e320-7c47-4ea9-8df8-e868326bc90b/pull/5/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='46'/>

Comment (weiji14) [inline notebook.ipynb:2]: <blockquote>Some parameters are fed to <code>predict_forecast</code> that ideally shouldn't need to be specified (like <code>seed</code> and <code>n_filters_factor</code>) and might seem contextually odd. They're used to locate the appropriate saved network. <em>This will be cleaned up in a future version</em>.</blockquote><p>Is the cleaned up version meant to be completed by this review? Or in a future as in after June 2024?</p><p> </p><p>Also, what does <code>n_filters_factor</code> do? Not immediately obvious. Edit: Seems to be documented in the 'Improving results' subsection below, maybe bring that up here.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/eds-book-gallery/67a1e320-7c47-4ea9-8df8-e868326bc90b/pull/5/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='49'/>

Comment (weiji14) [inline notebook.ipynb:2]: <p>State "NumPy (<code>.npy</code>) files", user might not know that numpy files = <code>.npy</code> files.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/eds-book-gallery/67a1e320-7c47-4ea9-8df8-e868326bc90b/pull/5/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='51'/>

Comment (weiji14) [inline notebook.ipynb:2]: <pre class="ql-syntax" spellcheck="false">Line #17.        fc = ds.sic_mean.isel(time=0).drop_vars("time").rename(dict(leadtime="time"))
Line #18.        fc['time'] = [pd.to_datetime(forecast_date) \
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; + dt.timedelta(days=int(e)) for e in fc.time.values]
</pre><p>These lines might benefit from some inline code comments to explain what is happening for users not adept at xarray/pandas.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/eds-book-gallery/67a1e320-7c47-4ea9-8df8-e868326bc90b/pull/5/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='57'/>

Comment (weiji14) [inline notebook.ipynb:2]: <pre class="ql-syntax" spellcheck="false">Line #21.        anim = xvid(fc, 15, figsize=4, mask=land_mask)
</pre><p>Is there a way to label the colorbar with something like "Sea-ice concentration fraction"? Maybe a feature request to the IceNet library?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/eds-book-gallery/67a1e320-7c47-4ea9-8df8-e868326bc90b/pull/5/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='57'/>

Comment (weiji14) [inline notebook.ipynb:2]: <p>More a note than a suggestion. Pytorch has an option to run in deterministic mode (see <a href="https://pytorch.org/docs/2.3/generated/torch.use_deterministic_algorithms.html" target="_blank">https://pytorch.org/docs/2.3/generated/torch.use_deterministic_algorithms.html</a>), is there something similar for Tensorflow?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/eds-book-gallery/67a1e320-7c47-4ea9-8df8-e868326bc90b/pull/5/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='61'/>

Comment (weiji14) [inline notebook.ipynb:2]: <p>Would prefer if the link to the license points to <a href="https://github.com/eds-book-gallery/67a1e320-7c47-4ea9-8df8-e868326bc90b/blob/main/LICENSE" target="_blank">https://github.com/eds-book-gallery/67a1e320-7c47-4ea9-8df8-e868326bc90b/blob/main/LICENSE</a> rather than <a href="https://github.com/alan-turing-institute/environmental-ds-book/blob/master/LICENSE.md" target="_blank">https://github.com/alan-turing-institute/environmental-ds-book/blob/master/LICENSE.md</a>, just in case the wider EDS license ever changes in the future, this notebook won't be accidentally re-licensed.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/eds-book-gallery/67a1e320-7c47-4ea9-8df8-e868326bc90b/pull/5/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='65'/>

Comment (bnubald) [reply notebook.ipynb:2]: <p>    This mentions three separate regions that are masked:</p><p>    1. Regions where sea ice does not form</p><p>    1. Land regions</p><p>    1. Polar hole (which as you mention is not observed)</p><p>    Have updated to include it as a region that is unobserved.</p>

Comment (bnubald) [reply notebook.ipynb:2]: <p>@annefou, happy with this change?</p><p>Current wording seems to be what's generally used in EDS Book, just seeking confirmation.</p>

Comment (bnubald) [reply notebook.ipynb:2]: <p>* Added definitions section</p><p>* Added link to original U-Net paper, and updated to be consistent with the original paper - U-Net.</p>

Comment (bnubald) [reply notebook.ipynb:2]: <p>Added acronyms section to header cell to cover this.</p>

Comment (bnubald) [reply notebook.ipynb:2]: <p>Added an explanation to variable mapping used, and the different options available in this implementation.</p>

Comment (bnubald) [reply notebook.ipynb:2]: <p>Added to acronyms section</p>

Comment (bnubald) [reply notebook.ipynb:2]: <p>* Added acronym</p><p>* Updated link as suggested.</p><p>* Added link to Apache Airflow.</p>

Comment (bnubald) [reply notebook.ipynb:2]: <p>Updated to <code>U-Net</code> for consistency to align with original U-Net paper. </p>

Comment (bnubald) [reply notebook.ipynb:2]: <p>This would be cleaned up in a future version after June 2024. Also, for <code>n_filters_factor</code>, added a brief description in parenthesis on first mention.</p>

Comment (bnubald) [reply notebook.ipynb:2]: <p>Added mention of Numpy as suggested.</p>

Comment (bnubald) [reply notebook.ipynb:2]: <p>Added option to do this, now the plots include the colourbar label.</p>

Comment (bnubald) [reply notebook.ipynb:2]: <p>This is something that was looked into, but some of the underlying methods required within Tensorflow did not support it. And, as with pytorch's approach, it would be deterministic only for the same hardware and come at a performance deficiency.</p><p> </p><p>More details here:</p><p><a href="https://github.com/icenet-ai/icenet/issues/8" target="_blank">https://github.com/icenet-ai/icenet/issues/8</a></p>

Comment (annefou) [reply notebook.ipynb:2]: We discussed this change and indeed the link to the license should point to the license file that is present in the notebook repository (here:  https://github.com/eds-book-gallery/67a1e320-7c47-4ea9-8df8-e868326bc90b/blob/main/LICENSE), rather than the EDSbook license file. I think we will need to update all the notebook repos accordingly. @acocac do you agree with this suggestion?

Thanks for raising this issue.

Comment (acocac) [reply notebook.ipynb:2]: Hi @acocac, thanks for reaching out. I agree we should point to the license file of the notebook repository. I have opened an issue to report this, https://github.com/alan-turing-institute/environmental-ds-book/issues/242. Thanks @weiji14.

Comment (bnubald) [reply notebook.ipynb:2]: <p>Updated in latest review based updates to point to local LICENSE file.</p>

Comment (acocac) [reply notebook.ipynb:2]: Thank you.

**Comment by acocac** [comment]
Merging into the main branch, however main changes were addressed in #6.
